taskcollector/TaskCollector.cpp

```diff
@@ -85,256 +97,239 @@ void FreeImageErrorHandler(FREE_IMAGE_FORMAT fif, const char *message) {
 // ----------------------------------------------------------
 
 class TaskCollector {
-	
+
 protected:
-	std::list<FIBITMAP *> chunks;
-	/*std::queue<FIBITMAP *> alphaChunks; */
-	
+	std::vector<bitmap_ptr> chunks;
+	std::vector<bitmap_ptr> alphaChunks;
 
 public:
-	
-	bool addImgFile(const char* pathName, int flag = 0)  {
-		FIBITMAP *img = GenericLoader(pathName, flag);
-		if (img == NULL) 
+	TaskCollector() = default;
+	TaskCollector(const TaskCollector&) = delete;
+	TaskCollector(TaskCollector &&other)

```

Comment:
You could use default move constructor here `TaskCollector(TaskCollector&&) = default`. However, I'm not sure MSVC support that yet.